documentation/retrospective/sprint2_retrospective.md

```diff
@@ -0,0 +1,128 @@
+
+In the following table, ratios are computed dividing the column value over the row value.
+
+| Story points | 2 | 5 | 8 |
+| --- | --- | --- | --- |
+| 2 | 1 | 0.56 | 0.47 |
+| 5 | 1.80 | 1 | 0.84 |
+| 8 | 2.13 | 1.18 | 1 |
+
+**Considerations on averages**
+
+Those results lead to the same considerations made for the estimated time, in the previous paragraph. On average, stories of 8 points required a total effort similar to the one of 5 points.
+
+
+### Considerations
+
+For what concerns single stories: 
+
+* _Story #6_ was overestimated. 8 story points were assigned to it, but it required an effort a bit bigger than a 2 point story. This consideration could be done also when the story was divided into tasks. Assigning an estimated time for each task, it turned out that the story could have been estimated as a 3 points story.
+* _Story #3_ was slightly underestimated, since, dividing it into tasks, it turned out to be more elaborated than expected. Probably, it could have a value of approximately 13 story points.
+* Remaining stories were estimated in a quite accurate way.
+
+## What caused errors in estimation
+
+* _Story #6_: working on this story, it turned out that the implementation logic was really similar to the logic of another story, so the effort required to implement this story was smaller than the one expected. Indeed, the other story worked as a sort of template to implement also this story.
+* _Story #3_: we underestimated this story because, when we did the estimation poker, it was not clear for us which could be the implications consequent to enabling access to parents. We didn't take into account some details: all the logic concerning creating a new account for parents, enabling it, send emails with initial credentials and perform the mandatory change of the password at the first access to the website.
+
+## Definition of Done
+
+* **Unit test**: rspec, rails tests;
+* **Code review**: at the end of each story (exploiting also pull requests);
+* **Code on VCS**: git on GitHub;
+* **Performed E2E test**: rspec.
+
+## Lessons learned
+
+Positive:
+
+* The first initial sprint, which was dedicated to learning a new language and framework, was really useful in order to work better in this sprint. This can be easily seen also comparing the number of stories committed and implemented in both sprints.
+* It's easier and faster to implement new features when you have a clear idea of how the logic should be done and you have implemented features with a similar implementation logic.
+
+Negative:
+
+* Sometimes, when a template or an interface for a web page is not clearly defined in advance all together, it could take more time to implement that webpage since different people could think about different implementations.
+* For what concerns Docker, we ecountered some problems. That is because initially it was not clear what we shoud have done and for this reason we wasted a lot of time in trying to do not required operations.
```

Comment:
* should
** unrequired

Reply:
** fatto

* Gabri, a cosa ti riferisci con should?

Reply:
Could è più adatto in quel contesto(secondo me)

Reply:
Ora ho capito gabriiii, correggo subito